packages/bot-web-ui/src/stores/run-panel-store.ts

```diff
@@ -514,15 +514,18 @@ export default class RunPanelStore {
         if (this.error_type === ErrorTypes.RECOVERABLE_ERRORS) {
             // Bot should indicate it started in below cases:
             // - When error happens it's a recoverable error
-            const { shouldRestartOnError, timeMachineEnabled } = this.dbot.interpreter.bot.tradeEngine.options;
-            const is_bot_recoverable = shouldRestartOnError || timeMachineEnabled;
-
-            if (is_bot_recoverable) {
-                this.error_type = undefined;
-                this.setContractStage(contract_stages.PURCHASE_SENT);
-            } else {
-                this.setIsRunning(false);
-                indicateBotStopped();
+            const trade_engine_options = this.dbot?.interpreter?.bot?.tradeEngine?.options;
+            if (trade_engine_options) {
+                const { shouldRestartOnError, timeMachineEnabled } = trade_engine_options;
```

Comment:
const shouldRestartOnError = trade_engine_options?.shouldRestartOnError || false;
const timeMachineEnabled = trade_engine_options?.timeMachineEnabled || false;

can we do this ?

Reply:
@rupato-deriv we already have a check if `is_bot_recoverable` = `true` then only the code will go, if `is_bot_recoverable` is `undefined` then condition `else` will be called. The suggestion is good, but in my opinion, it doesn't bring any improvement.😌
```
                const is_bot_recoverable = shouldRestartOnError || timeMachineEnabled;

                if (is_bot_recoverable) {
                    this.error_type = undefined;
                    this.setContractStage(contract_stages.PURCHASE_SENT);
                } else {
                    this.setIsRunning(false);
                    indicateBotStopped();
                }
```
probably would be better to set default values like this to make it safer:
`const { shouldRestartOnError = false, timeMachineEnabled = false } = trade_engine_options;`
what do you think about this option @rupato-deriv ?
Cos the issue we are getting when we try to access shouldRestartOnErrorin this line:
`const { shouldRestartOnError, timeMachineEnabled  } = trade_engine_options;`

Reply:
Yes, you can do this too. Looks good. @maryia-matskevich-deriv 
However i disagree that it does not bring any improvement. We are handling the case of `shouldRestartOnError`, `timeMachineEnabled` becoming undefined which would have caused error on the de-structuring. So in my opinion this suggestion would definitely bring some improvement atleast.

Reply:
but these cases work the same, aren't they?
`const { shouldRestartOnError = false, timeMachineEnabled = false } = trade_engine_options;` 
and 
```
const shouldRestartOnError = trade_engine_options?.shouldRestartOnError || false;
const timeMachineEnabled = trade_engine_options?.timeMachineEnabled || false;
```

Reply:
Yes i already mentioned in the first line you can do it in any of the two ways. 
Both are okay 👍.  @maryia-matskevich-deriv

Reply:
@rupato-deriv thank you for your suggestion, it's improved this bottleneck, it's great😌🤗